src/engines/dropped-frames-watcher.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+import FakeEventTarget from '../event/fake-event-target';
+import FakeEvent from '../event/fake-event';
+import {CustomEventType} from '../event/event-type';
+import getLogger from '../utils/logger';
+
+const EXCEEDED_MAX_FRAME_DROP: string = 'exceededmaxframerop';
+
+class DroppedFramesWatcher extends FakeEventTarget {
+  _droppedFramesInterval: ?number = null;
+  _lastDroppedFrames: number = 0;
+  _lastDecodedFrames: number = 0;
+  _lastTime: number = 0;
+  _mediaSourceAdapter: IMediaSourceAdapter;
+  _config: Object = {};
+  _videoElement: HTMLVideoElement;
+  static _logger: any = getLogger('droppedFramesWatcher');
+
+  constructor(mediaSourceAdapter: IMediaSourceAdapter, config: Object, videoElement: HTMLVideoElement) {
+    super();
+    this._mediaSourceAdapter = mediaSourceAdapter;
+    this._config = config;
+    this._videoElement = videoElement;
+    this._droppedFramesInterval = setInterval(() => this._maybeAdjustBitrateQuality(), this._config.fpsDroppedFramesInterval);
```

Comment:
`_maybeAdjustBitrateQuality` is not aproper name as it doesn't adjust bitrate(?~) but just checks for it